docs/concepts/package-types.md

```diff
@@ -0,0 +1,225 @@
+| `package`  | No              | Yes             | Yes            | Yes            | N/A                        | No                                           | Whatever specified in `main` field of `package.json` is used as entrypoint                                                                                                                                                                                                                       |
+| `view`     | Yes             | Yes             | Yes            | Yes            | N/A                        | No                                           | Whatever specified in `main` field of `package.json` is used as entrypoint; needs to default export React component for `modular start` to work                                                                                                                                                  |
+| `source`   | No              | No              | Yes            | Yes            | N/A                        | N/A - never built                            | N/A - don't need the concept of entrypoint since this package is never built                                                                                                                                                                                                                     |
+| `template` | No              | No              | Yes            | Yes            | Depends on the target type | N/A - never built                            | N/A - don't need the concept of entrypoint since this is a special type of package used by `modular add` which is never built directly                                                                                                                                                           |
+
+## App
+
+Modular `app`s are Typescript and React based Single Page Applications (SPA)
+built with [Webpack v5](https://webpack.js.org/) (by default) or
+[esbuild](https://esbuild.github.io/) (by turning on `useModularEsbuild` in
+[the configuration](../configuration.md)).
+
+Apps are generated by `modular add` using the
+[`modular-template-app`](https://github.com/jpmorganchase/modular/tree/main/packages/modular-template-app)
+[template](../concepts/templates.md) and need at least a `public/index.html`
+file and an entrypoint located in `src/index.tsx`, that typically `render`s a
+React component to the DOM, to work.
+
+Apps support pretty much all the functionalities from
+[Create React App v5](https://create-react-app.dev/docs/custom-templates), with
+some important differences:
+
```

Comment:
Maybe not something we need to do right now, but it would be good to have the _why_ for each point. Developers might be wondering why we don't have eject, why babel plugins are not supported, etc

Reply:
Added some "whys"